Content.Server/Forensics/Systems/ForensicsSystem.cs

```diff
@@ -106,12 +109,41 @@ private void OnAfterInteract(EntityUid uid, CleansForensicsComponent component,
             if (args.Handled || !args.CanReach)
                 return;
 
-            if (!TryComp<ForensicsComponent>(args.Target, out var forensicsComp))
+            args.Handled = StartClean(component, uid, args.User, args.Target, args.Used);
+        }
+
+        private void OnUtilityVerb(Entity<CleansForensicsComponent> entity, ref GetVerbsEvent<UtilityVerb> args)
+        {
+            if (!args.CanInteract || !args.CanAccess || !args.CanInteract)
                 return;
 
-            if((forensicsComp.DNAs.Count > 0 && forensicsComp.CanDnaBeCleaned) || (forensicsComp.Fingerprints.Count + forensicsComp.Fibers.Count > 0))
+            var user = args.User;
+            var target = args.Target;
+            var used = entity;
+
+            var verb = new UtilityVerb()
             {
-                var doAfterArgs = new DoAfterArgs(EntityManager, args.User, component.CleanDelay, new CleanForensicsDoAfterEvent(), uid, target: args.Target, used: args.Used)
+                Act = () =>
+                {
+                    StartClean(entity, entity, user, target, used);
+                },
+                IconEntity = GetNetEntity(entity),
+                Text = Loc.GetString(Loc.GetString("forensics-verb-text")),
+                Message = Loc.GetString(Loc.GetString("forensics-verb-message"))
+            };
+
+            args.Verbs.Add(verb);
+        }
+
+        public bool StartClean(CleansForensicsComponent cleanComp,
+                                EntityUid uid, EntityUid user, EntityUid? target, EntityUid used)
+        {
+            if (!TryComp<ForensicsComponent>(target, out var forensicsComp))
+                return false;
+
```

Comment:
These args a bit of a dog's breakfast and need cleaning up, use Entity<T> and also add some documenting to it.

Reply:
Sorry forgot to reply I think this is resolved with the latest commit!